lib/resque/scheduler.rb

```diff
@@ -339,15 +339,16 @@ def poll_sleep_loop
             sleep(remaining_sleep)
             handle_signals
           rescue Interrupt
-            if @shutdown
-              Resque.clean_schedules
-              release_master_lock
-            end
+            before_shutdown if @shutdown
             break
           end
         end
       end
 
+      def before_shutdown
```

Comment:
What if it's not shut down cleanly? Does it periodically refresh its presence as master?

:tropical_drink: :runner: :dash: